crates/accelerate/src/sabre/route.rs

```diff
@@ -383,13 +413,144 @@ impl<'a, 'b> RoutingState<'a, 'b> {
             }
         }
 
+        if let Some(DepthHeuristic { weight, scale }) = self.heuristic.depth {
+            let weight = match scale {
+                SetScaling::Constant => weight,
+                SetScaling::Size => {
+                    if self.front_layer.is_empty() {
+                        0.0
+                    } else {
+                        weight / (self.front_layer.len() as f64)
+                    }
+                }
+            };
+            let orig_depth = self.circuit_depth();
+
```

Comment:
Also, this code is very heavily duplicated from the main loop of the core Sabre loop.  I'm fine leaving it like this for its first introduction, but we should definitely consider what refactors we might make to avoid the duplication, since it's relatively complex code with a lot of different methods inlined into one place.

Reply:
I agree, I'll leave it as is for now, but we should definitely plan some refactoring to avoid duplication in the future.